File: library/src/main/java/com/snail/easyble/callback/ScanListener.java
package com.snail.easyble.callback;

import android.support.annotation.NonNull;

import com.snail.easyble.core.Device;

/**
 * 描述: 蓝牙扫描回调
 * 时间: 2018/5/9 00:08
 */
public interface ScanListener {
    /** 缺少定位权限 */
    int ERROR_LACK_LOCATION_PERMISSION = 0;
    /** 系统位置服务未开启 */
    int ERROR_LOCATION_SERVICE_CLOSED = 1;
    
    /**
     * 扫描开始
     */
    void onScanStart();

    /**
     * 扫描结束
     */
    void onScanStop();

    /**
     * 扫描结果
     * @param device 设备
     */
    void onScanResult(@NonNull Device device);

    /**
     * 扫描错误
     * @param errorCode {@link #ERROR_LACK_LOCATION_PERMISSION}, {@link #ERROR_LOCATION_SERVICE_CLOSED}
     */
    void onScanError(int errorCode, @NonNull String errorMsg);
}

File: library/src/main/java/com/snail/easyble/callback/ScanListener.kt
package com.snail.easyble.callback

import android.Manifest
import com.snail.easyble.core.Device

/**
 * 
 * date: 2018/5/9 00:08
 */
interface ScanListener {

    /**
     * 蓝牙搜索开始
     */
    fun onScanStart()

    /**
     * 蓝牙搜索停止
     */
    fun onScanStop()

    /**
     * 搜索到BLE设备
     */
    fun onScanResult(device: Device)

    /**
     * 搜索错误
     * 
     * @param errorCode [ERROR_LACK_LOCATION_PERMISSION], [ERROR_LOCATION_SERVICE_CLOSED]
     */
    fun onScanError(errorCode: Int, errorMsg: String)

    companion object {
        /** 缺少定位权限 [Manifest.permission.ACCESS_COARSE_LOCATION] 或者 [Manifest.permission.ACCESS_FINE_LOCATION] */
        const val ERROR_LACK_LOCATION_PERMISSION = 0
        /** 系统位置服务未开启 */
        const val ERROR_LOCATION_SERVICE_CLOSED = 1
    }
}
